fuzz/fuzz.c

```diff
@@ -412,9 +412,8 @@ int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
 
     ASS_Track *track = NULL;
 
-    // All return values but zero and -1 are reserved
     if (!init())
-        return 0;
+        exit(1);
```

Comment:
Can you elaborate on what benefit this change brings?
Whether the fuzzer is restarted or continues with the next sample shouldn’t make much difference considering no state but a fontconfig env var persists across samples anyway, except for the former being slower

Reply:
This is critical error and should never happen. If it does, fuzz.c needs to be fixed. It does not depend on fuzzing input.

Reply:
At least in AFL, a crash like this wouldn’t stop fuzzing either; instead it creates a bunch of bogus "crash" reports. Meaning if the fuzz environment is setup is so broken that we can't even init *(e.g. extremely restrictive memory limits)* we’d get a bunch of bogus reports.

Reply:
> At least in AFL, a crash like this wouldn’t stop fuzzing either

Good point, I've updated AFL code too.

> Meaning if the fuzz environment is setup is so broken that we can't even init (e.g. extremely restrictive memory limits) we’d get a bunch of bogus reports.

What do you mean by bogus? Those are very valid reports that fuzz environment setup is broken and cannot proceed.

Also note that any OOM would be caught earlier, because ASAN already aborts on any OOM. OSS-Fuzz is configured to 2G limit, I think by default ASAN defaults to 8G limit.